File: refactoring-practice-1/src/main/java/dojo/form_template_method/Site.java
package dojo.form_template_method;

public abstract class Site {

	final static double TAX_RATE = 0.27;
	
	private final int units;
	private final double rate;
	
	public Site(int units, double rate) {
		super();
		this.units = units;
		this.rate = rate;
	}
	
	public int getUnits() {
		return units;
	}
	
	public double getRate() {
		return rate;
	}
	
}

File: refactoring-practice-1/src/main/typescript/form_template_method/site.ts
/*
 * A leszármazott osztályok getBillableAmount metódusai nagyon hasonló dolgot csinálnak.
 * Készítsünk a számukra itt egy sablonfüggvényt, ami két absztrakt függvényt hív meg:
 * getBaseAmount és getTaxAmount!
 */
export abstract class Site {

    static TAX_RATE = 0.27;

    public constructor(private units: number, private rate: number) {
    }

    public getUnits() {
        return this.units;
    }

    public getRate() {
        return this.rate;
    }

}
